File: 2269-find-the-k-beauty-of-a-number/2269-find-the-k-beauty-of-a-number.py
class Solution:
    def divisorSubstrings(self, num: int, k: int) -> int:
        s = str(num)
        n = len(s)
        k_beauty = 0
        for i in range(n-k+1):
            sub_int = int(s[i:i+k])
            if sub_int and ((num % sub_int) == 0):
                k_beauty += 1
        return k_beauty

File: 2269-find-the-k-beauty-of-a-number/2269-find-the-k-beauty-of-a-number.cpp
class Solution {
public:
    int divisorSubstrings(int num, int k) {
        auto s = std::to_string(num);
        int n = s.length();
        int k_beauty = 0;
        int sub_int;
        std::string sub_str;
        for (int i=0; i<(n-k+1); i++) {
            sub_str = s.substr(i,k);
            sub_int = std::stoi(sub_str);
            if (sub_int and ((num % sub_int) == 0)) {
                k_beauty++;
            }
        }
        return k_beauty;
    }
};